### DISCUSSION TITLE
Stuck at FFMPEG - openssl not found (Windows 10)

### DISCUSSION THREAD

**Discussion by mike-frank** [AcademySoftwareFoundation/OpenRV#469] [Build Issues]
Hi all - I have the latest update that fixed Windows builds for PCRE2.

I'm now running into issues building RV_DEPS_FFMPEG.

> C:\OpenRV\_build\cmake\dependencies\RV_DEPS_FFMPEG.vcxproj" (default target) (1) ->
>        (CustomBuild target) ->
>          CUSTOMBUILD : error : openssl not found [C:\OpenRV\_build\cmake\dependencies\RV_DEPS_FFMPE
>        G.vcxproj]

Prior to building FFMPEG, I'm able to run `rvbuildt RV_DEPS_OPENSSL` and everything seems to compile correctly.

My config.log for FFMPEG looks like this. I did notice a mixture of windows vs linux style drive letters and paths, but I tried manually setting these all to `/c/` for one build (altered the paths in one of the cmake files) and it didn't seem to make a difference.

> ./configure --prefix='C:/OpenRV/_build/RV_DEPS_FFMPEG/install' --enable-shared --disable-static --disable-iconv --disable-outdevs --disable-programs --disable-large-tests --disable-vaapi --disable-doc --toolchain=msvc --disable-decoder=bink --disable-decoder=binkaudio_dct --disable-decoder=binkaudio_rdft --disable-decoder=vp9 --disable-decoder=vp9_cuvid --disable-decoder=vp9_mediacodec --disable-decoder=vp9_qsv --disable-decoder=vp9_rkmpp --disable-decoder=vp9_v4l2m2m --disable-decoder=dnxhd --disable-decoder=prores --disable-decoder=qtrle --disable-decoder=aac --disable-decoder=aac_at --disable-decoder=aac_fixed --disable-decoder=aac_latm --disable-decoder=dvvideo --disable-encoder=dnxhd --disable-encoder=prores --disable-encoder=qtrle --disable-encoder=aac --disable-encoder=aac_mf --disable-encoder=vp9_qsv --disable-encoder=vp9_vaapi --disable-encoder=dvvideo --disable-filter=geq --disable-parser=vp9 --disable-protocol=ffrtmpcrypt --disable-protocol=rtmpe --disable-protocol=rtmpte --extra-cflags=-I/c/OpenRV/_build/RV_DEPS_OPENSSL/install/include --extra-cflags=-I/c/OpenRV/_build/RV_DEPS_OPENSSL/install/include/openssl --extra-cxxflags=-I/c/OpenRV/_build/RV_DEPS_OPENSSL/install/include --extra-cflags='-IC:/OpenRV/_build/RV_DEPS_DAV1D/install/include' --extra-cflags='-IC:/OpenRV/_build/RV_DEPS_ZLIB/install/include' --extra-ldflags='-LIBPATH:/c/OpenRV/_build/RV_DEPS_OPENSSL/install/lib' --extra-ldflags='-LIBPATH:C:/OpenRV/_build/RV_DEPS_DAV1D/install/lib' --extra-ldflags='-LIBPATH:C:/OpenRV/_build/RV_DEPS_ZLIB/install/lib' --enable-openssl --enable-libdav1d
> ACLOCAL_PATH=C:/msys64/usr/share/aclocal/

Anyone have ideas how to get it to pick up openssl?

**Comment by cedrik-fuoco-adsk** [comment]
Could you try to do a fresh build (using rvbootstrap, or rvcfg and rvmk) and show the whole OpenRV log and the whole config.log from FFmpeg? @mike-frank

**Comment by mike-frank** [comment]
Hi @cedrik-fuoco-adsk thanks for taking a look. I wiped and re-pulled OpenRV just to make sure I was starting clean.

I forgot that I had to specify `-no-undefine` somewhere in one of the cmake files in order to get PCRE2 to build. I don't think the build got to FFMPEG on this run through (at least, there is no config.log present at C:\OpenRV\\_build\RV_DEPS_FFMPEG\src).

This might be a good place to start. I'm attaching the logs here - this config.log will be from PCRE2.

[rvbootstrap_init.log](https://github.com/AcademySoftwareFoundation/OpenRV/files/15418781/rvbootstrap_init.log)
[rvbuildlog.log](https://github.com/AcademySoftwareFoundation/OpenRV/files/15418782/rvbuildlog.log)
[config.log](https://github.com/AcademySoftwareFoundation/OpenRV/files/15418787/config.log)

**Comment by alexeiskachykhin** [comment]
I ran into the problem with the same symptoms on Windows 11 and the following helped:

1. https://github.com/microsoft/vcpkg/issues/9693
2. rvcfg
3. rvbuildt RV_DEPS_FFMPEG

**Reply by herronelou** [reply]
I'm working though similar issues, but for me the patch doesn't seem to help, I posted my progress here https://github.com/AcademySoftwareFoundation/OpenRV/issues/717
